### ISSUE TITLE
[Bug]: LlamaIndex Not Returning similarity_top_k Chunks with Azure Cognitive Search

### ISSUE THREAD

**Issue by FatimaKhan04** [run-llama/llama_index#16209]
### Bug Description

When using LlamaIndex with Azure Cognitive Search, setting similarity_top_k=20 and max_search_results=20 doesn't retrieve the expected 20 chunks. Instead, fewer chunks (e.g., 7-8) are returned.

### Version

 0.11.13

### Steps to Reproduce

1. Initialize AzureAISearchVectorStore with max_search_results=20.
2. Set similarity_top_k=20 in the retriever configuration.
3. Execute a query using the LlamaIndex query engine.
4. Observe that fewer than 20 chunks are returned.

**Comment by FatimaKhan04** [comment]
@dosu 
I'm already doing exactly as suggested by setting similarity_top_k=20 in my configuration, as mentioned in my original report. I've verified that similarity_top_k and max_search_results are set correctly. Additionally, I have confirmed that there are enough nodes available in my index (84 nodes in total), so availability isn't the issue.

It seems that despite these settings, the retrieval process still returns fewer chunks than expected. Could you please check if there's any other internal constraint that might be causing this?

**Comment by FatimaKhan04** [comment]
@dosu 
I've implemented all the recommendations you provided:

1. Verified Field Names: Confirmed that all field names in my code match those in my Azure Cognitive Search index.
2. Checked for Constraints: Ensured that there are no additional constraints like filters, node_ids, or doc_ids affecting the retrieval.
3. Configured Retriever: Set similarity_top_k=20 when creating the query engine.

Despite these changes, I'm still not receiving the expected number of chunks; the issue remains unresolved
